### PR TITLE
[PT FE] add boolean support for mul and add

### LINKED ISSUE #21971: [Good First Issue]: Support aten::add and aten::mul for boolean inputs
### Context

OpenVINO component responsible for support of PyTorch models is called as PyTorch Frontend (PT FE). PT FE converts a model represented as TorchScript model to a model in [OpenVINO opset](https://docs.openvino.ai/latest/openvino_docs_ops_opset.html).

### What needs to be done?

- Implement conversion rule and/or transformation to support the new operation.
- Implement operation tests in tests/layer_tests/pytorch_tests. Please consider different data types, but keep reasonable number of test cases

### Example Pull Requests

Refer to `aten::sum` [operation conversion](https://github.com/openvinotoolkit/openvino/blob/058f0206e2076a7b580eef06120850ce79625072/src/frontends/pytorch/src/op/sum.cpp#L21) for a similar case when we need to change logic when input is boolean.

### Resources

- [Contribution guide - start here!](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [What is OpenVINO?](https://github.com/openvinotoolkit/openvino#what-is-openvino-toolkit)
- [OpenVINO PyTorch Frontend](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/pytorch/README.md)
- [Blog post on contributing to OpenVINO](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [User documentation](https://docs.openvino.ai/)

### Contact points

@openvinotoolkit/openvino-pytorch-frontend-maintainers 

### Ticket

_No response_

### PR DESCRIPTION
**Pull request by pavel-esir**
## Details:
 - finalization of https://github.com/openvinotoolkit/openvino/pull/22045
 - for boolean inputs of Add and Mul use logical Or and And ops subsequently
 
### Tickets:
 - Closes #21971
 - _Ticket:_ CVS-133316

### CODE REVIEW

**Review by eaidova — APPROVED** [review]

**Review by eaidova — APPROVED** [review]

File: src/frontends/pytorch/src/op/add.cpp [hunk 1]
```diff
@@ -28,6 +29,16 @@ OutputVector translate_add_common(const NodeContext& context, bool inplace) {
         // Case when two lists gets concatenated
         PYTORCH_OP_CONVERSION_CHECK(false, "aten::add is used for concatenation of lists, not possible to convert");
     }
+
+    if (lhs.get_element_type() == element::boolean && rhs.get_element_type() == element::boolean) {
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/add.cpp:33]: Please also check dtype with `NodeContext::get_input_type()`. Like is done in:
https://github.com/openvinotoolkit/openvino/blob/fa43c7b364b6899a90da2b3a27d77461235a5071/src/frontends/pytorch/src/op/sum.cpp#L46

File: src/frontends/pytorch/src/op/mul.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+namespace pytorch {
+namespace op {
+
+using namespace ov::op;
+
+OutputVector translate_mul_common(const NodeContext& context, bool inplace) {
+    auto lhs = context.get_input(0);
+    auto rhs = context.get_input(1);
+
+    if (lhs.get_element_type() == ov::element::boolean && rhs.get_element_type() == ov::element::boolean) {
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/mul.cpp:21]: Same here

**Review by mvafin — CHANGES_REQUESTED** [review]

File: src/frontends/pytorch/src/op/add.cpp (hunk 1 above)
Comment (pavel-esir) [reply src/frontends/pytorch/src/op/add.cpp]: done

File: src/frontends/pytorch/src/op/mul.cpp (hunk 2 above)
Comment (pavel-esir) [reply src/frontends/pytorch/src/op/mul.cpp]: done

File: src/frontends/pytorch/src/op/mul.cpp [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+OutputVector translate_mul_common(const NodeContext& context, bool inplace) {
+    auto lhs = context.get_input(0);
+    auto rhs = context.get_input(1);
+    auto lhs_dtype = simplified_type_interpret(context.get_input_type(0));
+    auto rhs_dtype = simplified_type_interpret(context.get_input_type(1));
+
+    auto left_is_bool = lhs.get_element_type() == ov::element::boolean || (lhs_dtype.is<element::Type>() && lhs_dtype.as<element::Type>() == element::boolean);
+    auto right_is_bool = rhs.get_element_type() == ov::element::boolean || (rhs_dtype.is<element::Type>() && rhs_dtype.as<element::Type>() == element::boolean);
+
```
Comment (pavel-esir) [inline src/frontends/pytorch/src/op/mul.cpp:25]:
```suggestion
```

File: src/frontends/pytorch/src/op/add.cpp [hunk 4]
```diff
@@ -28,6 +29,19 @@ OutputVector translate_add_common(const NodeContext& context, bool inplace) {
         // Case when two lists gets concatenated
         PYTORCH_OP_CONVERSION_CHECK(false, "aten::add is used for concatenation of lists, not possible to convert");
     }
+
+    auto left_is_bool = lhs.get_element_type() == ov::element::boolean || (dtype0.is<element::Type>() && dtype0.as<element::Type>() == element::boolean);
+    auto right_is_bool = rhs.get_element_type() == ov::element::boolean || (dtype1.is<element::Type>() && dtype1.as<element::Type>() == element::boolean);
+
```
Comment (pavel-esir) [inline src/frontends/pytorch/src/op/add.cpp:35]:
```suggestion
```

**Review by mvafin — APPROVED** [review]

**Pull request merged**